utils/coreutils/techutils.go

```diff
@@ -191,6 +204,183 @@ func DetectedTechnologiesList() (technologies []string) {
 	return techStringsList
 }
 
+// If recursive is true, the search will not be limited to files in the root path.
+// If requestedTechs is empty, all technologies will be checked.
+// If excludePathPattern is not empty, files/directories that match the wildcard pattern will be excluded from the search.
+func DetectTechnologiesDescriptors(path string, recursive bool, requestedTechs []string, requestedDescriptors map[Technology][]string, excludePathPattern string) (technologiesDetected map[Technology]map[string][]string) {
+	filesList, err := fspatterns.ListFiles(path, recursive, false, true, true, excludePathPattern)
+	if err != nil {
+		return
+	}
+	workingDirectoryToIndicators, excludedTechAtWorkingDir := mapFilesToRelevantWorkingDirectories(filesList, requestedDescriptors)
+	strJson, err := json.MarshalIndent(workingDirectoryToIndicators, "", "  ")
```

Comment:
Added check, this error is only for creating the string representation, I don't think we should stop the flow in case of an error there.
Using MarshalIndent to list the files discovered more readable (map of paths)